0000-interledger-quoting-protocol-v2.md

```diff
@@ -0,0 +1,87 @@
+
+ILQPv2 is intended to be used alongside Transport Layer protocols. It uses a transfer condition with a special prefix (`ffffffffffffff` in hexadecimal or `__________` in base64url encoding). Other end-to-end protocols SHOULD ignore to avoid interfering with ILQPv2 and other protocols that use well-known conditions.
+
+## Model of Operation
+
+### Fixed Source Amount
+
+This details the steps for a sender to get a quote for a fixed source amount ("if I send this much, how much will the receiver get?"):
+
+1. The sender constructs an ILP packet with the receiver's ILP Address, an amount equal to the maximum value for unsigned 64-bit integers (18446744073709551615), and no data.
+2. The sender uses their ILP module to send the packet on a transfer with the amount set to the fixed source amount of their quote request.
+3. Connectors forward the payment to the receiver. (Note that connectors that recognize the special condition MAY "forward" the payment without actually reserving funds, since they know the payment will not be executed.)
+4. When the receiver is notified of the incoming transfer, they reject the transfer with an error message including the amount that arrived in the transfer and the amount of time they have before the transfer expires.
+5. Connectors relay the error message back to the sender.
+6. When the sender is notified that their test transfer was rejected, they inspect the error message to determine how much arrived at the receiver's ledger and how much time the receiver had to respond before the transfer would time out. The sender can then adjust the values before sending the actual payment.
+
+### Fixed Destination Amount
+
+The steps are the same as for Fixed Source Amount quotes, except that the transfer amount is set to an arbitrary probe value. When the sender gets the error response from the receiver, they compute the exchange rate based on the amount they sent and the amount that arrived. The sender then divides the fixed destination amount from the quote request by that exchange rate.
+
+## Specification
+
+### Condition
+
+ILQPv2 test payments use the condition:
+
+`fffffffffffffff8a5aa9fefdbed3fffffffffffffffffffffffffffffffffff`, encoded as hexadecimal.
+
+Or `__________ilqp_v2-0________________________`, encoded in Base64-URL.
+
+### Error Message Format
+
+**TODO: Define a binary format**
+
+Fields include:
+- Error code: either `F08` (which is currently unused) or `F99: Application Error`
+- Error name: `ILQPv2`
+- Amount that arrived in the final transfer to the receiver
+- Milliseconds between the final transfer expiry and when the receiver processed the notification
```

Comment:
Why do we care about that number?

Reply:
I put it in initially because I was trying to keep the [`destinationExpiryDuration`](https://github.com/interledgerjs/ilp/blob/b09395ecfe53d3a9eed1e13de84c33ed9b3df691/src/lib/ilqp.js#L119) feature. But isn't necessary if you assume that the time the receiver needs to fulfill and reject are the same, so the fact that they had enough time to reject means they'll have enough time to fulfill.